src/microphasing.rs

```diff
@@ -222,15 +318,37 @@ impl ObservationMatrix {
                 }
             }
 
-            fasta_writer.write(
-                &format!(
-                    "{}:{{\"offset\":{},\"af\":{:.2},\"variants\":{},\"somatic\":{}}}",
-                    transcript.id, offset, freq, n_variants, n_somatic
-                ),
-                None,
-                // restrict to window len (it could be that we insert too much above)
-                &seq[..window_len as usize]
-            )?;
+            let mut shaid = sha1::Sha1::new();
```

Comment:
This section needs some comments. It is quite cyclomatic as well. Maybe parts can be moved into separate functions/methods?